internal/controller/ipam/adapter/adapter.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2025
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapter
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kcm "github.com/K0rdent/kcm/api/v1beta1"
+)
+
+type IPAMAdapter interface {
+	BindAddress(ctx context.Context, config IPAMConfig, c client.Client) (kcm.ClusterIPAMProviderData, error)
```

Comment:
I would suggest changing the first return-argument to the reference

Reply:
reference makes sense only if ClusterIPAMProviderData is expected to by modified by append